lib/rubocop/cop/lint/interpolation_check.rb

```diff
@@ -35,6 +35,10 @@ def on_str(node)
 
         private
 
+        def regex_type?(parent)
+          parent&.dstr_type? || parent&.regexp_type?
```

Comment:
I don't quite get how those two checks imply that something's a regexp type. I'd say the method name is quite confusing.